### PR TITLE
[11.x] Allow sorting routes by precedence in artisan routes:list.

### PR DESCRIPTION
**Pull request by mathieutu**
Hello, 

By default, routes are sorted by `uri`.
However, it's not easy to see which routes will take precedence on the others (especially when routes are coming from a package).

This PR adds a `--sort=precedence` option to display the routes by their original order.

It also removes useless code, and adds a test to be sure we don't have any regression.

_PS: in the edge case where an array is passed like `['domain', 'precedence']`, it still will work as planned, as null will be compared to null in the sort callback._

Thanks.
Mathieu.

### CODE REVIEW

**Comment by mathieutu** [comment]
Tests are failing for a unrelated reason.
```
Symfony\Component\Mailer\Exception\UnsupportedSchemeException: The "" scheme is not supported; supported schemes for mailer "smtp" are: "smtp", "smtps".
```

File: src/Illuminate/Foundation/Console/RouteListCommand.php [hunk 1]
```diff
@@ -116,7 +116,11 @@ protected function getRoutes()
             return $this->getRouteInformation($route);
         })->filter()->all();
 
-        $routes = $this->sortRoutes($this->option('sort'), $routes);
+        if (($sort = $this->option('sort')) !== null) {
+            $routes = $this->sortRoutes($sort, $routes);
+        } else {
+            $routes = $this->sortRoutes('uri', $routes);
+        }
```
Comment (mathieutu) [inline src/Illuminate/Foundation/Console/RouteListCommand.php]: The option is by 'uri' by default, so it seems not needed, as it's never null. I added a test to be sure of it.

**Comment by taylorotwell** [comment]
Thanks!
